pkg/registration/status.go

```diff
@@ -13,12 +13,39 @@ const (
 	// System is not registered yet.
 	Unregistered
 
-	// System has an expired subscription.
-	Expired
+	// Set when an error occurred
+	Unknown
 )
 
+type statusRequest struct {
+	Hostname          string `json:"hostname"`
+	SystemInformation any    `json:"hwinfo"`
+}
+
 // Returns the registration status for the system pointed by the authorized
 // connection.
-func Status(conn connection.Connection) (StatusCode, error) {
+func Status(conn connection.Connection, hostname string, systemInformation any) (StatusCode, error) {
```

Comment:
I added here `hostname` and `systemInformation`. Both are required to update the system.